subscription.go

```diff
@@ -140,6 +152,35 @@ func (r *marathonClient) RegisterSubscription() error {
 	return nil
 }
 
+func (r *marathonClient) registerSSESubscription() error {
+	// step: get a member from the cluster
+	marathon, err := r.cluster.GetMember()
+	if err != nil {
+		return err
+	}
+	url := fmt.Sprintf("%s/%s", marathon, MARATHON_API_EVENT_STREAM)
+	stream, err := eventsource.Subscribe(url, "")
+	if err != nil {
+		glog.V(DEBUG_LEVEL).Infof("failed to connect to Marathon event stream, error: %s", err)
+		r.cluster.MarkDown()
+		// step: retry the request with another endpoint
+		return r.registerSSESubscription()
```

Comment:
What do you think about converting recursion to the loop? It might be a bit easier to read and to understand.